### PR TITLE
Fix modified due date message

### LINKED ISSUE #20288: Missing translations on changing due date of an issue
### Description

Changing the due date from an issue results in a status message with missing locales (see screenshot)

### Screenshots

image

### Gitea Version

1.17.0+rc1

### Can you reproduce the bug on the Gitea demo site?

Yes

### Operating System

ArchLinux

### Browser Version

Firefox 102

### PR DESCRIPTION
**Pull request by lhsazevedo**
Fix #20288

Preview:
image

### CODE REVIEW

**Review by delvh — APPROVED** [review]
Seems fine.
Although we could even think about switching the key around
(`modified the due date from %s to %s %s` sounds more natural to me)

**Comment by silverwind** [comment]
Agree, please swap "to" and "from" in `options/locale/locale_en-US.ini` and then swap the date arguments in the template.

**Review by delvh — APPROVED** [review]

**Review by Gusted — APPROVED** [review]

**Comment by wxiaoguang** [comment]
Maybe `%[1]s %[2]s %[3]s` is more clear than `%s %s %s`, and it's friendly for other languages.

File: templates/repo/issue/view_content/comments.tmpl [hunk 1]
```diff
@@ -340,7 +340,8 @@
 				</a>
 				<span class="text grey">
 					<a class="author" href="{{.Poster.HomeLink}}">{{.Poster.GetDisplayName}}</a>
-					{{$.locale.Tr "repo.issues.due_date_modified" (.Content | ParseDeadline) $createdStr | Safe}}
+					{{$parsedDeadline := .Content | ParseDeadline}}
```
Comment (6543) [inline templates/repo/issue/view_content/comments.tmpl:343]: just revert this ...

File: options/locale/locale_en-US.ini [hunk 2]
```diff
@@ -1420,7 +1420,7 @@ issues.due_date_form_remove = "Remove"
 issues.due_date_not_writer = "You need repository write access to update an issue's due date."
 issues.due_date_not_set = "No due date set."
 issues.due_date_added = "added the due date %s %s"
-issues.due_date_modified = "modified the due date to %s from %s %s"
+issues.due_date_modified = "modified the due date from %s to %s %s"
```
Comment (6543) [inline options/locale/locale_en-US.ini]:
```suggestion
issues.due_date_modified = "modified the due date from %[2]s to %[1]s %[3]s"
```

**Review by 6543 — COMMENTED** [review]
I think this does not need a template change

File: options/locale/locale_en-US.ini (hunk 2 above)
Comment (lhsazevedo) [reply options/locale/locale_en-US.ini]: One more advantage of `%[2]s to %[1]s` (keeping the original argument order in the template) is that it won't break other languages

**Comment by lhsazevedo** [comment]
The problem with the current template is that it passes only two arguments to `Tr`, and due_date_modified requires three:

Current:
```
{{$.locale.Tr "repo.issues.due_date_modified" (.Content | ParseDeadline) $createdStr | Safe}}
                                               ------------------------   ---------
                                                    1st arg (array)        2nd arg
```

Proposed fix:
```
{{$parsedDeadline := .Content | ParseDeadline}}
{{$.locale.Tr "repo.issues.due_date_modified" (index $parsedDeadline 0) (index $parsedDeadline 1) $createdStr | Safe}}
                                               -----------------------   -----------------------   ---------
                                                     1st argument                2nd arg            3rd arg
```

Note that now we're passing the array items in the original order as the from/to swap is being handled by indexed verbs in the translation format string (`%[2]s %[1]s %[3]s`).

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
Passin args as befor will help translators ... tanks

**Comment by tyroneyeh** [comment]
This need backport v1.17?

**Comment by silverwind** [comment]
It includes a translation so it'll be more work to backport including translations, but if someone wants to do it, it should be fine to backport.

**Comment by tyroneyeh** [comment]
Maybe just backport the template, it's easy without the locale

**Comment by zeripath** [comment]
to my eyes the locale change is compatible with the previous locale string so I think it's fine. 

Backport the whole thing.

**Comment by lhsazevedo** [comment]
I think it's fine as well, as the order was swapped in the locale line itself.

Can I open a PR to the 1.17 branch?

**Comment by zeripath** [comment]
yup, you may find the following script helpful

```sh
#!/bin/sh
PR="$1"
SHA="$2"
VERSION="$3"

if [ -z "$SHA" ]; then
    SHA=$(gh api /repos/go-gitea/gitea/pulls/$PR -q '.merge_commit_sha')
fi

if [ -z "$VERSION" ]; then
    VERSION="v1.17"
fi

echo git checkout origin/release/"$VERSION" -b backport-$PR-$VERSION
git checkout origin/release/"$VERSION" -b backport-$PR-$VERSION
git cherry-pick $SHA && git commit --amend && git push ihsazevedo backport-$PR-$VERSION && xdg-open https://github.com/go-gitea/gitea/compare/release/"$VERSION"...ihsazevedo:backport-$PR-$VERSION

```

**Pull request merged**